A35-grpc-gateway.md

```diff
@@ -0,0 +1,83 @@
+Integrate the grpc-gateway as a multi-language option to allow for REST endpoint support
+----
+* Author(s): Keith Adler (Talador12)
+* Approver: a11r
+* Status: Draft --- {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: Core (C++), Java, Python, and Go.
+* Last updated: 2020-09-14
+* Discussion at: https://groups.google.com/u/1/g/grpc-io/c/rEfqlLL9Emw (filled after thread exists)
+
+## Abstract
+
+Based on this [gRPC issue](https://github.com/grpc/grpc/issues/23979), the goal is to include the functionality of the [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) directly into the gRPC codebase across all languages. From @gnossen:
+
+```
+grpc-gateway is a separate binary/process from your server. It can be deployed alongside a server of any language, which (I assume) is what's happening in the gist you linked above. We have talked about in-library support for this sort of thing in the past. That way, you wouldn't have to orchestrate a separate gateway process. But that would be a pretty substantial effort.
+```
+
+As gRPC has grown in popularity and usage, an option to run in-library support would be a huge win for many teams. Instead of a sidecar, this gateway could be run alongside the gRPC server instance itself. As it would be a substantial effort, the background section will detail out the complexity of implementing across all languages and the steps required to start this process.
```

Comment:
Can you add a little more detail to this? I'm interested in your perspective on the usage of `grpc-gateway` out in the wild.